Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/utils/ActivityTracker.kt

```diff
@@ -7,8 +7,12 @@ import android.os.Bundle
 class ActivityTracker : ActivityLifecycleCallbacks {
     val resumedActivities = HashSet<Activity>()
     var backgroundExecutionAllowed = false
+    var currentActivity : Activity? = null
+        private set
 
-    override fun onActivityCreated(activity: Activity, bundle: Bundle?) {}
+    override fun onActivityCreated(activity: Activity, bundle: Bundle?) {
+        currentActivity = activity
```

Comment:
Resumed/paused is a better fit for what you're trying to track than created/destroyed.

For instance, if you go from EmulationActivity to SettingsActivity, EmulationActivity will become paused and SettingsActivity will become created, started, and resumed. Then when returning to EmulationActivity, SettingsActivity will become paused, stopped and destroyed, and EmulationActivity will become resumed. In this scenario, the current logic would cause currentActivity to be set to null instead of the EmulationActivity instance.